src/solver/simulation/common-eco-adq.cpp

```diff
@@ -58,12 +58,11 @@ static void RecalculDesEchangesMoyens(Data::Study& study,
 
         for (uint j = 0; j < study.areas.size(); ++j)
         {
-            auto* balance = callbackBalance(study.areas.byIndex[j]);
-            assert(balance && "Impossible to find the variable");
-            if (balance)
+            assert(balance[j] && "Impossible to find the variable");
+            if (balance[j])
             {
                 problem.SoldeMoyenHoraire[i]->SoldeMoyenDuPays[j]
-                  = balance->avgdata.hourly[decalPasDeTemps];
+                  = balance[j]->avgdata.hourly[decalPasDeTemps];
             }
             else
             {
```

Comment:
About the call to **retrieveAverageNTC** : 
```c++
int ret = retrieveAverageNTC(study, link, avgDirect, avgIndirect);
```
we could 2 calls to a single function : 
```c++
int ret = retrieveAverageNTC(study, link->directCapacities, avgDirect);
ret = retrieveAverageNTC(study, link->indirectCapacities, avgIndirect) && ret;
```
as the treatment is very similar for both direct and indirect capacities.
Or we can consider it is going too far in code factorization.

Reply:
That looks good, I will do that